build.gradle

```diff
@@ -24,16 +30,79 @@ dependencies {
     compile 'org.slf4j:slf4j-api:1.7.12'
     antlr 'org.antlr:antlr4:4.5'
 
-    testCompile 'org.codehaus.groovy:groovy-all:2.4.3'
-    testCompile 'org.spockframework:spock-core:1.0-groovy-2.4'
+    // using groovy 2.3 to avoid problems in eclipse
+    testCompile 'org.codehaus.groovy:groovy-all:2.3.11'
+    testCompile 'org.spockframework:spock-core:1.0-groovy-2.3'
+}
+
+//jar {
+//    manifest {
+//        attributes(
+//            'Built-Date': currentDate(),
+//            'Implementation-Version': version,
+//            'Implementation-Build': versioning.info.commit
+//        )
+//    }
+//}
+
+generateGrammarSource {
+    group "${project.name}"
+    description 'Generates the parser code based on the grammar'
+
+    outputDirectory = file("${outputDirectory}/org/asciidoctor/parser")
+}
+
+task pickAsciidocFile << {
+    group "${project.name}"
+    description 'Shows the result tree of a file'
+
+    def fileChooser = new JFileChooser(
+            currentDirectory: file('src/test/resources'),
+            dialogTitle: "Choose an excel file",
+            fileSelectionMode: JFileChooser.FILES_ONLY,
+            fileFilter: new FileNameExtensionFilter("AsciiDoc files", "asciidoc", "adoc", "ad", "asc", "txt")
+            )
+
+    fileChooser.showOpenDialog(null)
+    def file = fileChooser.selectedFile
+}
+
+task downloadExamples << {
+    group "${project.name}"
+    description 'Fetchs some AsciiDoc files from GitHub to test the grammar'
+
+    [
+        'asciidoctor-maven-examples',
+        'asciidoctor-maven-plugin',
+        'asciidoctor-gradle-plugin'
+    ]. each { projectName ->
+        download {
+            src "https://raw.githubusercontent.com/asciidoctor/${projectName}/master/README.adoc"
+            dest file("src/test/resources/${projectName}.adoc")
+        }
+
+    }
 }
 
-jar {
-    manifest {
-        attributes(
-            'Built-Date': currentDate(),
-            'Implementation-Version': version,
-            'Implementation-Build': versioning.info.commit
-        )
+file('src/test/resources').eachFileMatch(FileType.FILES, (~/.*.adoc/) ) { File f ->
```

Comment:
Do we have the test cases to verify the test rig result instead of checking them manually (I assume current it needs manual checking) ?